code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -927,13 +927,34 @@
 	hivecore_cooldown = FALSE
 	xeno_message(SPAN_XENOBOLDNOTICE("The weeds have recovered! A new hive core can be built!"),3,hivenumber)
 
-/datum/hive_status/proc/free_respawn(client/C)
+/datum/hive_status/proc/free_respawn(client/C, turf/spawning_turf)
 	stored_larva++
 	if(!hive_location || !hive_location.spawn_burrowed_larva(C.mob))
 		stored_larva--
+		respawn_on_turf(C, spawning_turf)
 	else
 		hive_ui.update_burrowed_larva()
 
+/datum/hive_status/proc/respawn_on_turf(client/C, turf/spawning_turf)
+	var/mob/living/carbon/xenomorph/larva/new_xeno = spawn_hivenumber_larva(spawning_turf, hivenumber)
+	if(isnull(new_xeno))
+		return FALSE
+
+	if(!SSticker.mode.transfer_xeno(C.mob, new_xeno))
+		qdel(new_xeno)
+		return FALSE
+
+	new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly burrows out of \the [spawning_turf]!"),
+	SPAN_XENODANGER("You burrow out of \the [spawning_turf] and awaken from your slumber. For the Hive!"))
```

Comment:
It would be better if you could override this message in the proc call because we aren't burrowing out of say a catwalk from slumber. Rather something more along the lines "The hive has no core! You manage to emerge from your old husk as a larva!"